types/share/share.go

```diff
@@ -1,14 +1,66 @@
 package share
 
 import (
+	"bytes"
+	"encoding/binary"
 	"fmt"
 
 	"github.com/celestiaorg/nmt"
 
 	"github.com/rollkit/celestia-openrpc/types/appconsts"
 	"github.com/rollkit/celestia-openrpc/types/core"
+	"github.com/rollkit/celestia-openrpc/types/namespace"
 )
 
+// ParseReservedBytes parses a byte slice of length
+// appconsts.CompactShareReservedBytes into a byteIndex.
+func ParseReservedBytes(reservedBytes []byte) (uint32, error) {
+	if len(reservedBytes) != appconsts.CompactShareReservedBytes {
+		return 0, fmt.Errorf("reserved bytes must be of length %d", appconsts.CompactShareReservedBytes)
+	}
+	byteIndex := binary.BigEndian.Uint32(reservedBytes)
+	if appconsts.ShareSize <= byteIndex {
+		return 0, fmt.Errorf("byteIndex must be less than share size %d", appconsts.ShareSize)
+	}
+	return byteIndex, nil
+}
+
+// InfoByte is a byte with the following structure: the first 7 bits are
+// reserved for version information in big endian form (initially `0000000`).
+// The last bit is a "sequence start indicator", that is `1` if this is the
+// first share of a sequence and `0` if this is a continuation share.
+type InfoByte byte
```

Comment:
why are these repeated in here and also in info_byte.go?

Reply:
Moved to `pkg share` 👍🏽